### PR TITLE
Task/xinwei/om 510

### PR DESCRIPTION
**Pull request by xinwei12**

### CODE REVIEW

File: transport.js [hunk 1]
```diff
@@ -57,15 +59,27 @@ module.exports = {
...
         var url         = this.createUrl(host, port)
-        Request.post(url,{formData: form},
-            function(err, resp, body){
-                if(resp.statusCode == 200){
-                    console.log("INFO - Successfully upload the report.");
-                }
-                else{
-                    console.log('Err - ' + body.toString());
+        var headers     = {
+            "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36",
```
Comment (mkotyk) [inline transport.js]: Yuck.  I assume this is to trick CloudFlare

File: vcs.js [hunk 2]
```diff
@@ -47,12 +47,14 @@ module.exports = {
...
+     * @param  {string} is_linux  - If the system is on linux or windows
      * @return {string|null}      - return project name as a string if found, null otherwise
      */
-    getProject: function(root_path) {
+    getProject: function(root_path, is_linux) {
         try{
-            var project_name = Common.systemSync('basename \`git rev-parse --show-toplevel\`', root_path);
-            return project_name;
+            var project_name = Common.systemSync('git rev-parse --show-toplevel', root_path);
+            if(is_linux){ return project_name.split('/').slice(-1)[0].replace('\n', ''); }
```
Comment (mkotyk) [inline vcs.js]: Could use path.sep to get the platform specific version of slash.  Then you don't need to pass `is_linux` around
`return project_name.split(path.sep).slice(-1)[0].replace('\n', '');`

https://nodejs.org/docs/latest/api/path.html#path_path_sep

File: report.js [hunk 3]
```diff
@@ -193,18 +195,20 @@ module.exports = {
         // get host name, parse raw data
         var host_name = Common.systemSync('hostname')
         var raw_data  = JSON.parse(data);
+        var islinux   = process.platform === "linux";
```
Comment (mkotyk) [inline report.js]: What happens on a Mac?

File: report.js [hunk 4]
```diff
@@ -193,18 +195,20 @@ module.exports = {
...
+        var end_name  = islinux ? '/package.json' : '\\package.json';
```
Comment (mkotyk) [inline report.js]: could use `path.sep` here

File: main.js [hunk 5]
```diff
@@ -55,10 +55,12 @@ function getResult(token, isSend = true, host = 'reshift.softwaresecured.com', p
     var token      = (args['token'] != null || args['token'] != 'undefined') ? args['token']: token;
     var host       = (args['host'] != null || args['host'] != 'undefined') ? args['host']: host;
     var port       = (args['port'] != null || args['port'] != 'undefined') ? args['port']: token;
+    var is_linux   = process.platform === "linux";
+    var sep        = is_linux ? '/' : '\\';
```
Comment (mkotyk) [inline main.js]: I would use path.sep

File: report.js [hunk 6]
```diff
@@ -34,7 +34,9 @@ module.exports = {
      * @return {string|null} - return audit result as a json if success, null otherwise
      */
     runAudit: function(root_path){
-        var data = Common.systemSync('ls', root_path);
+        var islinux = process.platform === "linux";
+        var command = islinux ? 'ls' : 'dir';
+        var data  = Common.systemSync(command, root_path);
```
Comment (mkotyk) [inline report.js]: This leaves a lot of room for failure (ie if the user has a custom ls or dir command).  I would use `fs.existAsync(root_path)`

https://nodejs.org/api/fs.html#fs_fs_existssync_path

File: vcs.js (hunk 2 above)
Comment (xinwei12) [reply vcs.js]: neat, i will change

**Pull request merged**